specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json

```diff
@@ -672,6 +672,146 @@
+          "description": "Name of the Event Grid namespace where the MQTT client was created or updated."
+        },
+        "state": {
+          "type": "string",
+          "description": "Configured state of the client. The value could be Enabled or Disabled."
+        },
+        "createdOn": {
+          "type": "string",
+          "description": "Time the client resource is created based on the provider's UTC time."
+        },
+        "updatedOn": {
+          "type": "string",
+          "description": "Time the client resource is last updated based on the provider's UTC time. If the client resource was never updated, this value is identical to the value of the 'createdOn' property."
+        },
+        "attributes": {
+          "type": "object",
+          "description": "The key-value attributes that are assigned to the client resource."
+        }
+      }
+    },
+    "MQTTClientDeletedEventProperties": {
+      "type": "object",
+      "description": "Schema of the Data property of an EventGridEvent for MQTT Client Deleted event.",
+      "properties": {
+        "clientAuthenticationName": {
+          "type": "string",
+          "description": "Unique identifier for the MQTT client that the client presents to the service for authentication. This case-sensitive string can be up to 128 characters long, and supports UTF-8 characters."
+        },
+        "clientName": {
+          "type": "string",
+          "description": "Name of the client resource in the Event Grid namespace."
+        },
+        "namespaceName": {
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was deleted."
+        }
+      }
+    },
+    "MQTTClientSessionConnectedEventProperties": {
+      "type": "object",
+      "description": "Schema of the Data property of an EventGridEvent for MQTT Client Session Connected event.",
+      "properties": {
+        "clientAuthenticationName": {
+          "type": "string",
+          "description": "Unique identifier for the MQTT client that the client presents to the service for authentication. This case-sensitive string can be up to 128 characters long, and supports UTF-8 characters."
+        },
+        "clientSessionName": {
+          "type": "string",
+          "description": "Unique identifier for the MQTT client's session. This case-sensitive string can be up to 128 characters long, and supports UTF-8 characters."
+        },
+        "namespaceName": {
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was connected"
+        },
+        "sequenceNumber": {
+          "type": "string",
+          "description": "A number that helps indicate order of MQTT client session connected or disconnected events. Latest event will have a sequence number that is higher than the previous event."
+        }
+      }
+    },
+    "MQTTClientSessionDisconnectedEventProperties": {
```

Comment:
Why do we define separate properties definitions for each type? If the goal is to reduce duplication we should create a base definition that has the common properties and add the additional properties in each derived definition.

Reply:
I wasn't sure about the best practice/standard here. We can definitely follow the model you have in your pr